src/app/components/BookFilters/BookFilters.jsx

```diff
@@ -1,38 +1,84 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { FilterIcon } from 'dgx-svg-icons';
+import {
+  contains as _contains,
+  findWhere as _findWhere,
+} from 'underscore';
 
 import Filter from './Filter';
-import Actions from '../../actions/BookActions';
 
 class BookFilters extends React.Component {
   constructor(props) {
     super(props);
 
+    this.state = {
+      // Create an array data structure of filter objects.
+      filters: this.props.filters.map(filter => ({
+        active: false,
+        id: filter.toLowerCase().split(' ').join('-'),
```

Comment:
In the event that this.props.filters is empty, the initial state assumes it's mappable and might throw an error if it cannot execute `.toLowerCase()` of undefined? Should we check if it is not empty prior to traversal?

Reply:
The default for `this.props.filters` is an empty array. If it's empty, map will return an empty array and won't go through the callback function.

Reply:
👍